src/CloseOpenIntervals.jl

```diff
@@ -60,16 +60,16 @@ See `?AbstractCloseOpen` for more information.
 SafeCloseOpen
 
 @inline Base.first(r::AbstractCloseOpen) = getfield(r,:start)
-@inline Base.first(r::AbstractCloseOpen{StaticInt{F}}) where {F} = StaticInt{F}()
-@inline Base.step(::AbstractCloseOpen) = One()
-@inline Base.last(r::AbstractCloseOpen{<:Integer,I}) where {I} = getfield(r,:upper) - oneunit(I)
-@inline Base.last(r::AbstractCloseOpen{<:Integer,StaticInt{L}}) where {L} = StaticInt{L}() - One()
+@inline Base.first(r::AbstractCloseOpen{StaticInt{F}}) where {F} = F
+@inline Base.step(::AbstractCloseOpen) = 1
+@inline Base.last(r::AbstractCloseOpen{<:Union{Int,StaticInt},I}) where {I} = getfield(r,:upper) - 1
+@inline Base.last(r::AbstractCloseOpen{<:Union{Int,StaticInt},StaticInt{L}}) where {L} = L - 1
```

Comment:
Note that this line returned a `StaticInt` before, but now returns an `Int`.

I do think that's fine though, `Base` methods should probably only be returning `Int`.